### PR TITLE
Update lifecylce commands for scabbard store 

### PR DESCRIPTION
**Pull request by agunde406**
Updates the skeleton commands to properly update the
service state.

Prepare: Add the scabbard service with prepared state and a
	context for two phase

Finalize: Update scabbard state to finalized and add an alarm
	to the context

Retire: Update the service state to retired

Purge: Remove the service and all associated consensus and commit
	state

### CODE REVIEW

File: services/scabbard/libscabbard/src/service/v3/arguments.rs [hunk 1]
```diff
@@ -17,15 +17,17 @@ use std::convert::TryFrom;
 use splinter::error::InvalidArgumentError;
 use splinter::service::ServiceId;
 
+use crate::store::service::ConsensusType;
```
Comment (peterschwarz) [inline services/scabbard/libscabbard/src/service/v3/arguments.rs:20]: Commit message should say "to use ConsensusType"

Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/arguments.rs:20]: Whoops, fixed

**Review by peterschwarz — APPROVED** [review]

File: services/scabbard/libscabbard/src/store/command/prepare_service.rs [hunk 2]
```diff
@@ -12,29 +12,96 @@
...
+            .with_epoch(1)
+            .with_participant_processes(service.peers().to_vec())
+            .with_state(Scabbard2pcState::WaitingForVoteRequest)
+            .with_this_process(service.service_id().clone().service_id())
+            .build()
+            .map_err(|err| InternalError::from_source(Box::new(err)))
+    }
+}
+
+fn get_coordinator(peers: Vec<ServiceId>) -> ServiceId {
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/store/command/prepare_service.rs:102]: Add API doc to this function pointing out the significance - that this is the algorithm for determining the coordinator.

File: services/scabbard/libscabbard/src/store/command/prepare_service.rs [hunk 3]
```diff
@@ -12,29 +12,96 @@
...
+    peers
+        .into_iter()
+        .min_by(|x, y| x.as_str().cmp(y.as_str()))
+        .expect("2PC always has at least one peer (self)")
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/store/command/prepare_service.rs:106]: Remove this panic

File: services/scabbard/libscabbard/src/store/command/prepare_service.rs (hunk 2 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/store/command/prepare_service.rs]: Added

File: services/scabbard/libscabbard/src/store/command/prepare_service.rs (hunk 3 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/store/command/prepare_service.rs]: removed

**Review by vaporos — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**